verify/db.go

```diff
@@ -26,6 +26,36 @@ func NewDB() *DB {
 	}
 }
 
+type DelegationsVerifier struct {
+	DB *DB
+}
+
+func (d *DelegationsVerifier) Unmarshal(b []byte, v interface{}, role string, minVersion int) error {
+	return d.DB.Unmarshal(b, v, role, minVersion)
+}
+
+// NewDelegationsVerifier returns a DelegationsVerifier that verifies delegations
+// of a given Targets. It reuses the DB struct to leverage verified keys, roles
+// unmarshals.
+func NewDelegationsVerifier(d *data.Delegations) (DelegationsVerifier, error) {
```

Comment:
Does this need to be exported by the `verify` package? Alternatively, could it be private to the client (the only place it's used)? I think all we'd need is a way to make a delegated role pass the validation of `AddRole()`.

Reply:
Yes, we could add an exportable AddDelegatedRole and have everything in the client code. Current implem avoids adding delegated roles to DB and using a shared DB to check delegations. Having a separate struct with only Unmarshal enforces that the DelegatedVerifier is used for delegations: parent verifies childs. When we add Delegations in the repo  we can reuse this verifier to check that they are valid